src/main.rs

```diff
@@ -92,29 +108,51 @@ impl HanoiError {
 ///    * `Action`: if the input was well formed
 ///    * `Hanoi::UnknownCommand`: otherwise
 fn parse_action(input: &str) -> Result<Action,HanoiError> {
-    unimplemented!()
+    if input == "q" {
+      return Ok(Action::Quit);
+    }
+    let mut chars = input.chars();
+    let from = chars.next().and_then(char_to_peg);
+    let to = chars.next().and_then(char_to_peg);
+    match (from, to, chars.next()) {
+        (Some(from_peg), Some(to_peg), None) => Ok(Action::Move(Move::new(from_peg, to_peg))),
+        _ => Err(HanoiError::UnknownCommand),
+    }
 }
 
 impl State {
 
     /// Creates a Towers of Hanoi game with `disks` disks in a single tower
     fn new(disks: u8) -> State {
-        unimplemented!()
+        let mut vec = Vec::new();
+        for i in 0..disks { 
+            vec.push(Disk(disks - i));
+        }
```

Comment:
Definitely works! You can also take of the advantage of the fact that `0..disks` is `Iterator<i32>` and use `Iterator::collect` to pull the items in the iterator into a vector.